src/linear_algebra/dimension.lean

```diff
@@ -462,10 +462,20 @@ begin
     exact le_antisymm w₁ w₂, }
 end
 
+/-- Given two basis indexed by `ι` and `ι'` of an `R`-module, where `R` satisfies the invariant
+basis number property, an equiv `ι ≃ ι' `. -/
+def equiv_of_basis (v : basis ι R M) (v' : basis ι' R M) : ι ≃ ι' :=
+nonempty.some (cardinal.lift_mk_eq.1 (cardinal.lift_max.2 (mk_eq_mk_of_basis v v')))
+
 theorem mk_eq_mk_of_basis' {ι' : Type w} (v : basis ι R M) (v' : basis ι' R M) :
   #ι = #ι' :=
 cardinal.lift_inj.1 $ mk_eq_mk_of_basis v v'
 
+/-- Given two basis indexed by `ι` and `ι'` of an `R`-module, where `R` satisfies the invariant
+basis number property, an equiv `ι ≃ ι' `. -/
+def equiv_of_basis' {ι' : Type w} (v : basis ι R M) (v' : basis ι' R M) : ι ≃ ι' :=
+nonempty.some (cardinal.eq.1 (mk_eq_mk_of_basis' v v'))
```

Comment:
I don't see the difference between this one and `equiv_of_basis`, could you explain it (in the docstring)?

Reply:
`ι` and `ι'` are in the same universe (this is why there are is `lift` in this one). I am not sure I really understand this, but it's the same fr `mk_eq_mk_of_basis` and `mk_eq_mk_of_basis'`. I will clarify this.

Reply:
Indeed the second one is strictly less general than the first one, and so it is useless.

Reply:
That's what I thought :grin: The reason that there are two versions of the cardinality statement is because you can't actually express `#ι = #ι'` unless `ι` and `ι'` have different universes: you need to lift them to the same universe first.